scripts/sign_rproc_fw.py

```diff
@@ -0,0 +1,520 @@
+        assert img.get_machine_arch() in ["ARM"]
+
+        # Need to reopen the file to get the raw data
+        with open(inputf, 'rb') as f:
+            bin_img = f.read()
+        size = len(bin_img)
+        align_64b = size % 8
+        if align_64b:
+            size += 8 - align_64b
+
+        images_size.extend(size.to_bytes(4, 'little'))
+        s_header.img_length += size
+        f.close()
+
+        # Store image type information
+        # TODO support other format
+        bin_type = ENUM_BINARY_TYPE['ELF']
+        images_type += bin_type.to_bytes(1, 'little')
+
+        # Compute the hash table and add it to TLV blob
+        hash_table = SegmentHash(img)
+        hash_tlv.extend(hash_table.get_table())
+
+    # Add image information
+    # The 'IMGTYPE' contains a byte array of the image type (ENUM_BINARY_TYPE).
+    # The 'IMGSIZE' contains a byte array of the size (32-bit) of each image.
+    tlv.add('NUM_IMG', len(args.inf).to_bytes(1, 'little'))
+    tlv.add('IMGTYPE', bytearray(images_type))
+    tlv.add('IMGSIZE', bytearray(images_size))
+
+    # Add hash type information in TLV blob
+    # The 'HASHTYPE' TLV contains a byte associated to ENUM_HASH_TYPE.
+    hash_type = ENUM_HASH_TYPE['SHA256']
+    tlv.add('HASHTYPE', hash_type.to_bytes(1, 'little'))
+
+    # Add hash table information in TLV blob
+    # The HASHTABLE TLV contains a byte array containing all the elf segment
+    # with associated hash.
+    tlv.add('HASHTABLE', hash_tlv)
+
+    # Add optional key information to TLV
+    if args.key_infof:
+        with open(args.key_infof, 'rb') as f:
+            key_info = f.read()
+        tlv.add('PKEYINFO', key_info)
+
+    # Compute the Trailer containing TLV (with 64 bit alignment)
+    trailer_buff = tlv.get()
+    s_header.tlv_length = len(trailer_buff)
+
+    # TODO: create a function for alignment
+    align_64b = 8 - (s_header.tlv_length % 8)
+    if align_64b:
+        s_header.tlv_length += 8 - align_64b
+        trailer_buff += bytearray(8 - align_64b)
+
+    dump_buffer(trailer_buff, name='TRAILER', indent="\t")
+
+    # Signature chunk
+    sign_size = sig_size_type.get(ENUM_SIGNATURE_TYPE[args.key_type],
+                                  lambda: "Invalid sign type")(key)
+    s_header.sign_length = sign_size
+
+    # Construct the Header
+    header = s_header.get_packed()
+
+    # Generate signature
+    signer = Signature.get(ENUM_SIGNATURE_TYPE[args.key_type])(key)
+
+    signer.hash_compute(header)
+    signer.hash_compute(trailer_buff)
+    signature = signer.sign()
+    if len(signature) != sign_size:
+        raise Exception(("Actual signature length is not equal to ",
+                         "the computed one: {} != {}".
+                         format(len(signature), sign_size)))
+
+    s_header.dump()
+
+    with open(args.outf, 'wb') as f:
+        f.write(header)
+        f.write(signature)
+        f.write(bytearray(sign_size - s_header.sign_length))
```

Comment:
Is it possible `sign_size != s_header.sign_length`? This line can be removed IMO.